staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller.go

```diff
@@ -397,6 +418,8 @@ func (c *Controller) syncPool(ctx context.Context, poolName string) error {
 
 		logger.V(5).Info("Removing resource slices after pool removal", "obsoleteSlices", klog.KObjSlice(obsoleteSlices), "slices", klog.KObjSlice(slices), "numDevices", len(pool.Devices))
 		obsoleteSlices = append(obsoleteSlices, slices...)
+		// No need to create or update the slices.
+		slices = nil
```

Comment:
we set the slice to nil to avoid re-creating it? (Not sure if this logic is correct)

Reply:
I think it's right. All of these slices are obsolete, so we shouldn't try updating them below. Probably wasn't reached in practice before, but presumably you now have unit tests for it, right?

Reply:
yes. I discovered this seems to be a problem through unit testing, so I raised it. It can be tested from this case ``single ResourceSlice Existing but with mismatching resource pool name, so we just delete it``.